rust-iml.spec

```diff
@@ -74,6 +74,8 @@ mkdir -p %{buildroot}%{_presetdir}
 cp 00-rust-iml-agent.preset %{buildroot}%{_presetdir}
 mkdir -p %{buildroot}%{_sysconfdir}/iml/
 cp settings.conf %{buildroot}%{_sysconfdir}/iml/iml-agent.conf
+mkdir -p %{buildroot}%{_sysconfdir}/bash_completion.d
+%{buildroot}%{_bindir}/iml shell-completion bash -e iml -o %{buildroot}%{_sysconfdir}/bash_completion.d/iml
```

Comment:
Should we allow users to do this instead of doing it for them?

Reply:
AFAIK, completions are usually packaged alone with the binaries. So it could just work.

Reply:
Would probably be better as a standalone RPM with a dependency on this package.

Reply:
Zsh users are free to do it themselves :)

Reply:
I.E. 

```
%package cli-bash-completions
Requires: rust-iml-cli
...
```

Reply:
Done!

Reply:
```suggestion
```

This should be done in the `%post` scriptlet of `cli-bash-completion` instead.

Reply:
But then the file will not removed on removal, will it?

Reply:
And if in post-scripts, then I don't see a point in a separate package at all.

Reply:
> And if in post-scripts, then I don't see a point in a separate package at all.

Why? it gives the user a way to decide whether they want completions installed or not.

Reply:
> But then the file will not removed on removal, will it?

`%postun`?

Reply:
If the file is created in scripts, it will be hard to figure out which package is responsible for it, right?
And then it will be almost the same as letting users execute `iml shell-completion ...` on their own, even in home dir,
In addition, they will have to install the `bash-completion` package anyway.

Also generating the completion file when building a package should be more reliable, because we can catch any issue earlier and there will be no risk when installing the package.

Note that bash completions are shipped along with the executables (as well as man pages, etc.), except Django :)
```
# rpm -qf /etc/bash_completion.d/
filesystem-3.2-25.el7.x86_64
yum-utils-1.1.31-52.el7.noarch
iprutils-2.4.17.1-2.el7.x86_64
createrepo-0.9.9-28.el7.noarch

# rpm -qf /usr/share/bash-completion
glib2-2.56.1-5.el7.x86_64
yum-3.4.3-163.el7.centos.noarch
python-django-bash-completion-1.11.27-1.el7.noarch
```

So there are four options in the UX descending order (IMHO):
1. Same package as the executable.
2. Separate package (+ one for zsh, etc.), this might be a good idea.
3. Separate package with scripts (which will have to be coordinated with each other).
4. Manual `iml shell-completion ...`.

Open question is Docker packaging, but at least number 4 should will do, and 3 seems the worst.

Reply:
I'm a bit weary of executing the binary during build time, but I suppose hitting issues earlier is a fair point. Let's go with #2 including separate packages for ZSH.

I think we will need to revisit how we do CLI in docker generally, but not something for this issue.